pyomo/contrib/doe/doe.py

```diff
@@ -25,1255 +25,1169 @@
-            # Initialize with L in L_initial
-            if self.L_initial is not None:
-                model.L_ele = pyo.Var(
-                    model.regression_parameters,
-                    model.regression_parameters,
-                    initialize=init_cho,
-                )
-            # or initialize with the identity matrix
-            else:
-                model.L_ele = pyo.Var(
-                    model.regression_parameters,
-                    model.regression_parameters,
-                    initialize=identity_matrix,
-                )
-
-            # loop over parameter name
-            for i, c in enumerate(model.regression_parameters):
-                for j, d in enumerate(model.regression_parameters):
-                    # fix the 0 half of L matrix to be 0.0
-                    if i < j:
-                        model.L_ele[c, d].fix(0.0)
-                    # Give LB to the diagonal entries
-                    if self.L_LB:
-                        if c == d:
-                            model.L_ele[c, d].setlb(self.L_LB)
+        # TODO: Allow Params for `unknown_parameters` and `experiment_inputs`
+        #       May need to make a new converter Param to Var that allows non-string names/references to be passed
+        #       Waiting on updates to the parmest params_to_vars utility function.....
+
+        # Run base model to get initialized model and check model function
+        for comp in model.base_model.experiment_inputs:
+            comp.fix()
+
+        try:
+            res = self.solver.solve(model.base_model, tee=self.tee)
+            assert res.solver.termination_condition == "optimal"
+            self.logger.info("Model from experiment solved.")
+        except:
+            raise RuntimeError(
+                "Model from experiment did not solve appropriately. Make sure the model is well-posed."
+            )
 
-        # jacobian rule
-        def jacobian_rule(m, p, n):
-            """
-            m: Pyomo model
-            p: parameter
-            n: response
-            """
-            cuid = pyo.ComponentUID(n)
-            var_up = cuid.find_component_on(m.block[self.scenario_num[p][0]])
-            var_lo = cuid.find_component_on(m.block[self.scenario_num[p][1]])
-            if self.scale_nominal_param_value:
-                return (
-                    m.sensitivity_jacobian[p, n]
-                    == (var_up - var_lo)
-                    / self.eps_abs[p]
-                    * self.param[p]
-                    * self.scale_constant_value
-                )
+        for comp in model.base_model.experiment_inputs:
+            comp.unfix()
+
+        # Generate blocks for finite difference scenarios
+        def build_block_scenarios(b, s):
+            # Generate model for the finite difference scenario
+            b.transfer_attributes_from(model.base_model.clone())
```

Comment:
This construction rule is referencing a model object outside of its scope which can be dangerous. You can access the model containing the block data object using `m = b.model()`.